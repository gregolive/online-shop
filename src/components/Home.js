import '../styles/Home.css';

const Home = () => {
  return (
    <section className='Home'>
      <div className='HomeWrapper'>
        <h2 className='HeroText'>Your one stop shop for high quality espresso tools!</h2>
        <button type='button' className='Button ShopButton'>Shop now</button>
      </div>
    </section>
  );
};

export default Home;
