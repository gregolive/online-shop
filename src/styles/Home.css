.Home {
  background: center / cover no-repeat url(../img/banner.jpg);
  padding: 0;
  position: relative;
}

.HomeWrapper {
  background: rgba(0, 0, 0, 0.6);
  min-height: 100%;
  width: 100%;
  padding: 0 5vw;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  gap: 2rem;
  position: absolute;
  text-align: center;
  box-sizing: border-box;
}

.HeroText {
  color: var(--primary);
  font-size: clamp(1.5rem, 1.5rem + 1vw, 2.4rem);
}

.HomeWrapper > * > .Button {
  font-size: 1.4rem;
}
