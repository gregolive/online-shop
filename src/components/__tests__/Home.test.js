import { render, screen } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import Home from '../Home';

test('renders hero text', () => {
  render(<MemoryRouter><Home /></MemoryRouter>);
  expect(screen.getByRole('heading').textContent).toMatch(/Your one stop shop for high quality espresso tools!/);
});

test('renders shop now button', () => {
  render(<MemoryRouter><Home /></MemoryRouter>);
  expect(screen.getByRole('button').textContent).toMatch(/Shop now/);
});

it('renders consistently', () => {
  const { container } = render(<MemoryRouter><Home /></MemoryRouter>);
  expect(container).toMatchSnapshot();
});
