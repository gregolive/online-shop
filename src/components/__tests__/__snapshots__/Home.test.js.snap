// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders consistently 1`] = `
<div>
  <section
    class="Home"
  >
    <div
      class="HomeWrapper"
    >
      <h2
        class="HeroText"
      >
        Your one stop shop for high quality espresso tools!
      </h2>
      <a
        href="/shop"
      >
        <button
          class="Button PrimaryButton"
          type="button"
        >
          Shop now
        </button>
      </a>
    </div>
  </section>
</div>
`;
